### PR TITLE
Arrowheads are bad for you

### PR DESCRIPTION
**Pull request by ericbrooke**
Look at the code in the Arrowhead exercise. It currently fails to put error conditions near their conditions. Refactor it so that it isn't terrible anymore.

### CODE REVIEW

File: arrowhead/arrowhead.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class Arrowhead
+  # Error Messages
+  BAD_REGION = "Unknown region, please provide a valid region."
```
Comment (oharace) [inline arrowhead/arrowhead.rb]: Good idea setting these error messages aside.

File: arrowhead/arrowhead.rb [hunk 2]
```diff
@@ -15,19 +19,20 @@ class Arrowhead
     },
   }
 
-  # FIXME: I don't have time to deal with this.
+  def self.valid_region?(region)
```
Comment (oharace) [inline arrowhead/arrowhead.rb]: These methods are simple and to-the-point. Awesome!

File: arrowhead/arrowhead.rb [hunk 3]
```diff
@@ -15,19 +19,20 @@ class Arrowhead
...
-      if shapes.include? shape
-        arrowhead = shapes[shape]
-        "You have a(n) '#{arrowhead}' arrowhead. Probably priceless."
-      else
-        raise "Unknown shape value. Are you sure you know what you're talking about?"
-      end
-    else
-      raise "Unknown region, please provide a valid region."
-    end
+    return raise BAD_REGION unless valid_region?(region)
```
Comment (oharace) [inline arrowhead/arrowhead.rb]: I like the way you simplified this method. Much more readable.

**Comment by oharace** [comment]
Excellent work! You're ready for the next exercise.

**Comment by ericbrooke** [comment]
@oharace Thanks :-)

File: arrowhead/arrowhead.rb (hunk 3 above)
Comment (paulghaddad) [inline arrowhead/arrowhead.rb]: You don't need the `return` if you're raising an exception.

File: arrowhead/arrowhead.rb [hunk 4]
```diff
@@ -15,19 +19,20 @@ class Arrowhead
...
+    shapes = CLASSIFICATIONS[region]
```
Comment (paulghaddad) [inline arrowhead/arrowhead.rb]: Consider extracting this as a method.

**Comment by paulghaddad** [comment]
@ericbrooke Nice work on this! Feel free to move on!

**Pull request closed without merging**